src/main/java/org/apache/platypus/server/luceneserver/QueryNodeMapper.java

```diff
@@ -106,6 +115,40 @@ private DisjunctionMaxQuery getDisjunctionMaxQuery(org.apache.platypus.server.gr
         return new DisjunctionMaxQuery(disjuncts, disjunctionMaxQuery.getTieBreakerMultiplier());
     }
 
+    private Query getMatchQuery(MatchQuery matchQuery, IndexState state) {
+        Analyzer analyzer = isAnalyzerDefined(matchQuery.getAnalyzer())
+                ? getAnalyzer(matchQuery.getAnalyzer())
+                : state.searchAnalyzer;
+
+        MultiFieldQueryParser multiFieldQueryParser = new MultiFieldQueryParser(null, analyzer);
```

Comment:
Would the input queries not have boosts? Seems like MultiFieldQueryParser supports boosts on fields as well.

Reply:
I will add a query level boost separately (and per-field boost for multi-match).